go/vt/vtgate/evalengine/eval.go

```diff
@@ -129,7 +129,7 @@ func evalToSQLValueWithType(e eval, resultType sqltypes.Type) sqltypes.Value {
 		case *evalDecimal:
 			return sqltypes.MakeTrusted(resultType, e.dec.FormatMySQL(e.length))
 		}
-	default:
+	case e != nil:
```

Comment:
Nice! Can you also add a unit test here?

https://github.com/vitessio/vitess/blob/ab76091df9b3a4ed7df8a09dbe5fa78121a8e1ba/go/vt/vtgate/evalengine/api_arithmetic_test.go#L738-L749